tests/test_databinder.py

```diff
@@ -447,6 +447,50 @@ def run_query(input_filenames=None):
         with pytest.raises(SyntaxError):
             _load_ServiceXSpec(path)
 
+    with open(path := (tmp_path / "python.yaml"), "w") as f:
+        f.write("""
+General:
+  OutputFormat: root-ttree
+  Delivery: LocalCache
+
+Sample:
+  - Name: ttH3
+    RucioDID: user.kchoi:user.kchoi.fcnc_tHq_ML.ttH.v11
+    Query: !UprootRaw |
+                [{"treename": "nominal"}]
+  - Name: ttH4
+    Dataset: !Rucio user.kchoi:user.kchoi.fcnc_tHq_ML.ttH.v11
+    Query: !UprootRaw '[{"treename": "nominal"}]'
+  - Name: ttH5
+    Dataset: !FileList ["/path/to/file1.root", "/path/to/file2.root"]
+    Query: !UprootRaw '[{"treename": "nominal"}]'
+    """)
+        f.flush()
+        with pytest.raises(RuntimeError):
+            _load_ServiceXSpec(path)
+
+    with open(path := (tmp_path / "python.yaml"), "w") as f:
+        f.write("""
+General:
+  OutputFormat: root-ttree
+  Delivery: LocalCache
+
+Sample:
+  - Name: ttH3
+    RucioDID: user.kchoi:user.kchoi.fcnc_tHq_ML.ttH.v11
+    Query: !UprootRaw |
+                [{"treename": "nominal"}]
+  - Name: ttH5
+    Dataset: !FileList ["/path/to/file2.root", "/path/to/file1.root"]
+    Query: !UprootRaw '[{"treename": "nominal"}]'
+  - Name: ttH5
+    Dataset: !FileList ["/path/to/file1.root", "/path/to/file2.root"]
+    Query: !UprootRaw '[{"treename": "nominal"}]'
+    """)
+        f.flush()
+        with pytest.raises(RuntimeError):
+            _load_ServiceXSpec(path)
+
```

Comment:
Is there a test that checks that different NFiles or different queries for the same dataset avoid an exception?

Reply:
Added new test cases